tests/ui/pattern/usefulness/consts-opaque.rs

```diff
@@ -51,56 +49,47 @@ fn main() {
 
     match BAR {
         Bar => {}
-        BAR => {} // should not be emitting unreachable warning
+        BAR => {}
         //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
-        //~| ERROR unreachable pattern
         _ => {}
         //~^ ERROR unreachable pattern
     }
 
     match BAR {
         BAR => {}
         //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
-        Bar => {} // should not be emitting unreachable warning
-        //~^ ERROR unreachable pattern
+        Bar => {}
         _ => {}
         //~^ ERROR unreachable pattern
     }
 
     match BAR {
         BAR => {}
         //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
-        BAR => {} // should not be emitting unreachable warning
+        BAR => {}
         //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
-        //~| ERROR unreachable pattern
-        _ => {} // should not be emitting unreachable warning
-        //~^ ERROR unreachable pattern
+        _ => {}
     }
 
     match BAZ {
         BAZ => {}
         //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
-        Baz::Baz1 => {} // should not be emitting unreachable warning
-        //~^ ERROR unreachable pattern
+        Baz::Baz1 => {}
         _ => {}
-        //~^ ERROR unreachable pattern
     }
 
     match BAZ {
         Baz::Baz1 => {}
         BAZ => {}
         //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
         _ => {}
-        //~^ ERROR unreachable pattern
     }
 
     match BAZ {
         BAZ => {}
         //~^ ERROR must be annotated with `#[derive(PartialEq, Eq)]`
-        Baz::Baz2 => {} // should not be emitting unreachable warning
-        //~^ ERROR unreachable pattern
-        _ => {} // should not be emitting unreachable warning
-        //~^ ERROR unreachable pattern
+        Baz::Baz2 => {}
+        _ => {}
```

Comment:
This one actually is unreachable, right? BAZ is Baz1, and then we also cover Baz2, and then all variants are covered.

But I guess it's fine not to also emit that error here, seems better than emitting too many errors in other cases.

Reply:
That would mean using structural match semantics on `Baz`, which is the whole point we're trying to avoid, right? I think treating `BAZ` as opaque is correct here

Reply:
Ah, that's fair... I forgot that we require PartialEq *and* Eq to be derived. Yeah that makes sense. This will likely change when we clean that up but that's for the future.

Reply:
Looking forward to the cleanups :D